core/blockchain/impl/block_tree_impl.cpp

```diff
@@ -913,18 +918,10 @@ namespace kagome::blockchain {
 
   void BlockTreeImpl::collectDescendants(
       std::shared_ptr<TreeNode> node,
-      std::vector<std::pair<primitives::BlockHash, primitives::BlockNumber>>
-          &container) {
-    // avoid deep recursion
-    while (node->children.size() == 1) {
-      container.emplace_back(node->block_hash, node->depth);
-      node = node->children.front();
-    }
-
-    // collect descendants' hashes recursively
-    for (const auto &child : node->children) {
+      std::list<std::shared_ptr<TreeNode>> &container) {
+    container.emplace_front(node);
+    for (auto child : node->children) {
```

Comment:
Why? This change is canceling avoiding of redundant recursion

Reply:
Changed to iterative.